.github/workflows/main.yml

```diff
@@ -0,0 +1,587 @@
+name: build_macos
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '20[2-9][0-9]-[0-9][0-9]-[0-9][0-9]'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+    branches:
+      - master
+
+jobs:
+  macos-deps-build:
+    name: '01 - Build macOS dependencies'
+    runs-on: [macos-11]
+    strategy:
+      matrix:
+        arch: ['x86_64', 'arm64']
+    env:
+      MACOSX_DEPLOYMENT_TARGET: '10.13'
+      CACHE_REVISION: '02'
+      LIBPNG_VERSION: '1.6.37'
+      LIBPNG_HASH: '505e70834d35383537b6491e7ae8641f1a4bed1876dbfe361201fc80868d88ca'
+      LIBOPUS_VERSION: '2021-05-12'
+      LIBOPUS_HASH: 'dfd6c88aaa54a03a61434c413e30c217eb98f1d5'
+      LIBOGG_VERSION: '1.3.5'
+      LIBOGG_HASH: 'c4d91be36fc8e54deae7575241e03f4211eb102afb3fc0775fbbc1b740016705'
+      LIBVORBIS_VERSION: '1.3.7'
+      LIBVORBIS_HASH: 'b33cc4934322bcbf6efcbacf49e3ca01aadbea4114ec9589d1b1e9d20f72954b'
+      LIBVPX_VERSION: '1.10.0'
+      LIBVPX_HASH: '85803ccbdbdd7a3b03d930187cb055f1353596969c1f92ebec2db839fa4f834a'
+      LIBX264_VERSION: 'r3059'
+      LIBX264_HASH: 'b684ebe04a6f80f8207a57940a1fa00e25274f81'
+      LIBTHEORA_VERSION: '1.1.1'
+      LIBTHEORA_HASH: 'f36da409947aa2b3dcc6af0a8c2e3144bc19db2ed547d64e9171c59c66561c61'
+      LIBLAME_VERSION: '3.100'
+      LIBLAME_HASH: 'ddfe36cab873794038ae2c1210557ad34857a4b6bdc515785d1da9e175b1da1e'
+      MBEDTLS_VERSION: '2.24.0'
+      MBEDTLS_HASH: 'b5a779b5f36d5fc4cba55faa410685f89128702423ad07b36c5665441a06a5f3'
+      LIBSRT_VERSION: '1.4.1'
+      LIBSRT_HASH: 'e80ca1cd0711b9c70882c12ec365cda1ba852e1ce8acd43161a21a04de0cbf14'
+      FFMPEG_VERSION: '4.2.3'
+      FFMPEG_HASH: '9df6c90aed1337634c1fb026fb01c154c29c82a64ea71291ff2da9aacb9aad31'
+      SPEEXDSP_VERSION: '1.2.0'
+      SPEEXDSP_HASH: 'd7032f607e8913c019b190c2bccc36ea73fc36718ee38b5cdfc4e4c0a04ce9a4'
+      LIBJANSSON_VERSION: '2.13.1'
+      LIBJANSSON_HASH: 'f4f377da17b10201a60c1108613e78ee15df6b12016b116b6de42209f47a474f'
+      LIBLUAJIT_VERSION: '2.1'
+      LIBLUAJIT_HASH: 'ec6edc5c39c25e4eb3fca51b753f9995e97215da'
+      LIBFREETYPE_VERSION: '2.10.4'
+      LIBFREETYPE_HASH: '86a854d8905b19698bbc8f23b860bc104246ce4854dcea8e3b0fb21284f75784'
+      LIBRNNOISE_VERSION: '2020-07-28'
+      LIBRNNOISE_HASH: '90ec41ef659fd82cfec2103e9bb7fc235e9ea66c'
+      BLOCKED_FORMULAS: 'ant gradle maven selenium-server-standalone llvm gcc postgresql openjdk sox libsndfile flac libvorbis opusfile libogg php gd freetype fontconfig webp libpng lame libtiff opus'
```

Comment:
Should this be sorted alphabetically so there's some order to it?

Reply:
Sorted by build order which is also in parts dependant on quirks in how FFmpeg dependencies need to be built (i.e. MbedTLS and SRT).